internal/pkgmanager/download.go

```diff
@@ -5,69 +5,44 @@ import (
 	"io"
 	"net/http"
 	"os"
-	"path/filepath"
-	"strings"
-
-	"github.com/ossf/package-analysis/internal/utils"
 )
 
 /*
-downloadToDirectory downloads a file from the given URL to the given directory.
-On successful download, the full path to the downloaded file is returned.
-
-fileName argument is required with no default value, and the hash of the file
-is appended to the given filename. If an error occurs during hashing, then the
-original filename is used.
-*/
-func downloadToDirectory(dir string, url string, fileName string) (string, error) {
-	filePath := filepath.Join(dir, fileName)
+downloadToPath creates (and/or truncates) a file at the given path, then writes
+contents of whatever is at the given URL to that given file using downloadToFile,
+and finally closes the file.
 
-	if err := downloadToPath(filePath, url); err != nil {
-		return "", err
-	}
+If any error occurs, the created file is removed.
 
-	hash, err := utils.HashFile(filePath)
-	if err != nil {
-		// Error is ignored and the path with the original filename is returned
-		// We treat hashing as 'best-effort' rather than a strictly necessary part of the function
-		return filePath, nil
+Callers must ensure that path and url are nonempty, otherwise the function will panic.
+*/
+func downloadToPath(path string, url string) error {
```

Comment:
nit: this can be shortened to `downloadToPath(path, url string)`

Reply:
done